src/modules/fstab/main.py

```diff
@@ -151,23 +151,27 @@ def generate_crypttab_line_info(self, partition):
         if not mapper_name or not luks_uuid:
             return None
 
-        password = "/crypto_keyfile.bin"
         crypttab_options = self.crypttab_options
+        if os.path.isfile(os.path.join(self.root_mount_point, "crypto_keyfile.bin")):
+            password = "/crypto_keyfile.bin"
+        else:
+            password = "none"
 
-        # Set crypttab password for partition to none and remove crypttab options
         # if root partition was not encrypted
-        if any([p["mountPoint"] == "/"
-                and "luksMapperName" not in p
-                for p in self.partitions]):
+        if any([p["mountPoint"] == "/" and
+               "luksMapperName" not in p for p in self.partitions]):
+            crypttab_options = ""
+       # on root partition when /boot is unencrypted
+        elif partition["mountPoint"] == "/" and
+             any([p["mountPoint"] == "/boot"
+                 and "luksMapperName" not in p for p in self.partitions]):
+            crypttab_options = ""
             password = "none"
             crypttab_options = ""
-        # on root partition when /boot is unencrypted
-        elif partition["mountPoint"] == "/":
-            if any([p["mountPoint"] == "/boot"
-                   and "luksMapperName" not in p
-                   for p in self.partitions]):
-                password = "none"
-                crypttab_options = ""
+        # Setup keyfile
+        else:
+            password = "/crypto_keyfile.bin"
+            crypttab_options = self.crypttab_options
```

Comment:
I think these lines and the two directly above are no longer needed.

Reply:
I agree.
I missed it, GitHub's change review interface could use some work ;-)